cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -65,6 +66,60 @@ inline void knn_merge_parts(value_t* in_keys,
     in_keys, in_values, out_keys, out_values, n_samples, n_parts, k, stream, translations);
 }
 
+/**
+ * Performs a k-select across row partitioned index/distance
+ * matrices formatted like the following:
+ * row1: k0, k1, k2
+ * row2: k0, k1, k2
+ * row3: k0, k1, k2
+ * row1: k0, k1, k2
+ * row2: k0, k1, k2
+ * row3: k0, k1, k2
+ *
+ * etc...
+ *
+ * @tparam idx_t
```

Comment:
Is this an internal function?

Reply:
No, not internal. The mnmg knn uses this. I'd like to eventually move it to `raft::matrix`, but probably not until the top-k is moved there.